proc-macros/src/api_def.rs

```diff
@@ -154,7 +154,9 @@ impl ApiMethodAttrs {
 	/// Tries to merge another `ApiMethodAttrs` within this one. Returns an error if there is an
 	/// overlap in the attributes.
 	// TODO: span
-	fn try_merge(&mut self, other: ApiMethodAttrs) -> syn::parse::Result<()> {
+	fn try_merge(&mut self, other: ApiMethodAttrs)
+	//  -> syn::parse::Result<()>
+	{
 		if let Some(method) = other.method {
 			if self.method.is_some() {
 				// TODO: return Err(())
```

Comment:
@niklasad1 is there something here that is blocking?
I "fixed" the clippy issue but this is NOT a good option.

I understand that you planned on returning an Error but clippy is currently right in the sense that we don't need to return an Error if we don't actually return somewhere.

So the real fix here would be to implement the TODO and bring back the Error checking.

Reply:
Nothing really,

I think you could just return `return Err(syn::Error::new(method.span(), "Duplicate method attribute found"))` but I think it's already checked in another method before this.

FWIW, not really an expert on this code it was written by Pierre for a while ago :)

Reply:
Ok, thanks for the input @niklasad1.
I definitely don't want to just remove a `Result<>` and miss this error, although we don#t have it today 🤫
So returning this Err now, we take a small risk that it breaks downstream.
@tomaka can you advise?

Reply:
I don't think we should listen to clippy if it's suggesting stupid things such as removing the `Result`.

Reply:
I agree but I mean't brint back the `Result` as it was before and just replace `todo` and return `return Err(syn::Error::new(method.span(), "Duplicate method attribute found"))`

Reply:
I guess clippy does not read the comments so its conclusion was not off but I agree we should not remove the Result<>.